test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenTransactSpecs.java

```diff
@@ -496,6 +502,51 @@ public HapiApiSpec uniqueTokenTxnAccountBalance() {
 				);
 	}
 
+	public HapiApiSpec uniqueTokenTxnAccountBalancesForTreasury() {
+		return defaultHapiSpec("UniqueTokenTxnAccountBalancesForTreasury")
+				.given(
+						newKeyNamed("supplyKeyA"),
+						newKeyNamed("signingKeyTreasury"),
+						newKeyNamed("signingKeyNewTreasury"),
+						cryptoCreate(TOKEN_TREASURY).key("signingKeyTreasury"),
+						cryptoCreate("newTreasury").key("signingKeyNewTreasury"),
+						tokenCreate(A_TOKEN)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.initialSupply(0)
+								.supplyKey("supplyKeyA")
+								.treasury(TOKEN_TREASURY),
+						mintToken(A_TOKEN, List.of(ByteString.copyFromUtf8("memo"))),
+						tokenAssociate("newTreasury", A_TOKEN)
+				).when(
+						cryptoTransfer(
+								movingUnique(1, A_TOKEN).between(TOKEN_TREASURY, "newTreasury")
+						).signedBy("signingKeyTreasury", DEFAULT_PAYER).via("cryptoTransferTxn"),
+						cryptoTransfer(
+								movingUnique(1, A_TOKEN).between("newTreasury", TOKEN_TREASURY)
+						).signedBy("signingKeyTreasury", DEFAULT_PAYER).via("cryptoTransferTxn")
```

Comment:
I am not sure why we are changing the nft back and forth. If the transfer does not work at all, the then logic will pass again so we are not testing the transfer this way.
We are simply testing the getAccountBalance, getTokenNftInfo, getAccountNftInfos and getTokenNftInfos queries for mint. Now that that's the case lets remove the crypto transfers form this particular test, move the test to management specs (since we test the mint) and add new tests for the transfers. We have a lot of edge cases to cover there